src/headless/converse-chatboxes.js

```diff
@@ -765,6 +767,32 @@ converse.plugins.add('converse-chatboxes', {
                 }
             },
 
+            /**
+             * Responsible for setting the editable attribute of messages.
+             * If _converse.lmc_only_for_last is true, then only the last
+             * message sent from me will be editable.
+             * @method _converse.ChatBox#set_editable
+             * @memberOf _converse.ChatBox
+             * @param { Object } attrs An object containing message attributes.
+             * @param { String } send_time - time when the message was sent
+             */
+            set_editable (attrs, send_time) {
+                if (u.isEmptyMessage(attrs) || attrs.sender !== 'me') {
+                    return;
+                }
+                if (!_converse.lmc_only_for_last) {
+                    attrs.editable = true;
+                    return;
+                }
+                if (send_time > this.get('time_last_sent_msg')) {
+                    this.set({'time_last_sent_msg': send_time});
+                    this.messages.where({'editable': true}).forEach(msg => {
+                        msg.save({'editable': false});
+                    });
```

Comment:
What I like about having an `editable` attribute is that we can also later use it in semi-anon MUCs together with [XEP-0421](https://xmpp.org/extensions/xep-0421.html) to determine whether a message from the same nick is actually from the same person and therefore whether they're allowed to edit it.

If `time_last_sent_msg` is `true`, then at any one time only one message should have `editable` to `true`, which means you can use `this.messages.findWhere` which stops iterating once the first match has been found.

Note, `findWhere` returns the message itself, not an array.